__tests__/components/MnemonicInput.spec.ts

```diff
@@ -13,53 +13,482 @@
+            };
+
+            // Act:
+            const wrapper = getMnemonicInputWrapper(props);
+            const component = wrapper.vm as MnemonicInputTs;
+            component.handleWordsArray(item);
+
+            // Assert:
+            expect(component.wordsArray).toStrictEqual(expectedWordsArray);
+            expect(wrapper.emitted('handle-words')).toBeTruthy();
+            expect(wrapper.emitted('handle-words')[0]).toStrictEqual([expectedWordsArray]);
+        };
+
+        test('add word to wordsArray when it is provided as argument', () => {
+            // Arrange:
+            const item = 'AnotherWord';
+            const expectedWordsArray = ['word', 'word', 'anotherword'];
+
+            // Act + Assert:
+            runHandleWordsArrayTest(item, expectedWordsArray);
+        });
+
+        test('remove last word from wordsArray when argument is missing', () => {
+            // Arrange:
+            const item = null;
+            const expectedWordsArray = ['word'];
+
+            // Act + Assert:
+            runHandleWordsArrayTest(item, expectedWordsArray);
+        });
+    });
+
+    describe('handlePaste()', () => {
+        test('set seed from clipboard event', () => {
+            // Arrange:
+            const expectedClipboardData = 'clipboard data';
+            const props = {
+                seed: '',
+            };
+            const mockHandleSeed = jest.fn();
+            const clipboardEvent = {
+                clipboardData: {
+                    getData: () => ({
+                        toString: () => expectedClipboardData,
+                    }),
+                },
+            };
+
+            // Act:
+            const wrapper = getMnemonicInputWrapper(props);
+            const component = wrapper.vm as MnemonicInputTs;
+            component['handleSeed'] = mockHandleSeed;
+            component.handlePaste(clipboardEvent as any);
+
+            // Assert:
+            expect(mockHandleSeed).toBeCalledWith(expectedClipboardData);
+        });
+    });
+
+    describe('handleClickPaste()', () => {
+        test('set seed from navigator clipboard', async () => {
+            // Arrange:
+            const expectedClipboardData = 'clipboard data';
+            const props = {
+                seed: '',
+            };
+            const mockHandleSeed = jest.fn();
+            (window as any).__defineGetter__('navigator', () => ({
+                clipboard: {
+                    readText: () => Promise.resolve(expectedClipboardData),
+                },
+            }));
+
+            // Act:
+            const wrapper = getMnemonicInputWrapper(props);
+            const component = wrapper.vm as MnemonicInputTs;
+            component['handleSeed'] = mockHandleSeed;
+            await component.handleClickPaste();
+
+            // Assert:
+            expect(mockHandleSeed).toBeCalledWith(expectedClipboardData);
+        });
+    });
+
+    describe('handleSeed()', () => {
```

Comment:
generally it's an antipattern to test private methods directly. since they're implementation details, it's better to test side-effects / state changes of public methods as opposed to testing privates directly.